File: CardGameAPI/CardGameAPI/Models/Game.cs
using CardGame.Repositories;
using NLog;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace CardGame.Models
{
    public class Game
    {
        [Key]
        public Guid ID { get; set; }
        public string Name { get; set; }
        public bool Active { get; set; } = false;

        [NotMapped]
        public GameEngine Engine { get; set; }

        public Game() { }

        public Game(Game game, Logger logger, IQueryable<Card> cards)
        {
            ID = game.ID;
            Name = game.Name;
            Active = game.Active;
            Engine = new GameEngine(logger, cards);
        }
    }

    // Used to let a Player join an existing Game
    public class PlayerGame
    {
        public Guid GameID { get; set; }
        public Guid PlayerID { get; set; }
    }
}

File: CardGameAPI/CardGameAPI/CardGameUI/src/app/shared/models/game.ts
import {ICard} from './card';
import {IGamePlayer, IUser} from './player';

export interface IGame
{
    id: string;
    name: string;
    active: boolean;
    engine: IGameEngine;
}

export interface IGameEngine 
{
    gamePlayers: IUser[];
    CurrentGamePlayerID?: string;
    cards: ICard[];
    started: boolean;
    finished: boolean;
}

export interface IPlayerGame
{
    gameID: string;
    playerID: string;
}

export interface IGameState
{
    gamePlayers: IGamePlayer[];
    currentGamePlayerId?: string;
    cards: ICard[];
}
